fabric-network/src/impl/wallet/hsmx509identity.ts

```diff
@@ -83,21 +86,21 @@ export class HsmX509Provider implements IdentityProvider {
 		const data: HsmX509IdentityDataV1 = {
 			credentials: {
 				certificate: identity.credentials.certificate,
+				privateKey: identity.credentials.privateKey
 			},
 			mspId: identity.mspId,
 			type: 'HSM-X.509',
-			version: 1,
+			version: 2,
 		};
 		return data;
 	}
 
 	public async getUserContext(identity: HsmX509Identity, name: string): Promise<User> {
-		const user: User = new User(name);
+		const user = new User(name);
 		user.setCryptoSuite(this.cryptoSuite);
-
-		const publicKey: ICryptoKey = await this.cryptoSuite.importKey(identity.credentials.certificate);
-		const privateKeyObj: ICryptoKey = await this.cryptoSuite.getKey(publicKey.getSKI());
-		await user.setEnrollment(privateKeyObj, identity.credentials.certificate.toString(), identity.mspId);
+		const handle = Buffer.from(identity.credentials.privateKey, 'hex');
+		const privateKey = new Pkcs11EcdsaKey({priv: handle}, this.cryptoSuite.getKeySize());
```

Comment:
If a v1 HSM identity has been loaded from a wallet, the `privateKey` will likely be an empty string (depending on how you implement `fromJson()` above). Having it fail is fine as we can't magic up credentials we don't have but it might be worth understanding what the failure will look like from a user perspective to ensure it's not misleading for them.

But definitely the error should occur either here or when signing happens, *not* within `fromJson()` since it's perfectly reasonable for a user to get a v1 HSM identity from their wallet, add correct `privateKey` details and then put it good identity back into the wallet. It's only trying to use an identity containing bad credentials that should fail.